### PR TITLE
Add time picker for range selection

### PR DESCRIPTION
**Pull request by MilkBiscuit**

### CODE REVIEW

File: Sources/WWCalendarTimeSelector.swift [hunk 1]
```diff
@@ -999,7 +1033,11 @@ open class WWCalendarTimeSelector: UIViewController, UITableViewDelegate, UITabl
         case .multiple:
             del?.WWCalendarTimeSelectorDone?(picker, dates: multipleDates)
         case .range:
```
Comment (ygweric) [inline Sources/WWCalendarTimeSelector.swift]: Can you split enum value *range* to *dateRange* and *timeRange*, so we don't need *if* block in *swift* block

File: Sources/WWCalendarTimeSelector.swift [hunk 2]
```diff
@@ -2285,6 +2356,7 @@ internal protocol WWClockProtocol: NSObjectProtocol {
     func WWClockSwitchAMPM(isAM: Bool, isPM: Bool)
     func WWClockSetHourMilitary(_ hour: Int)
     func WWClockSetMinute(_ minute: Int)
+    func didChangeTimeStateHourFlag(showingHour: Bool)
```
Comment (ygweric) [inline Sources/WWCalendarTimeSelector.swift]: rename **didChangeTimeStateHourFlag** to **WWClockSetTimeStateHour**

File: Sources/WWCalendarTimeSelector.swift [hunk 3]
```diff
@@ -1187,8 +1225,17 @@ open class WWCalendarTimeSelector: UIViewController, UITableViewDelegate, UITabl
         monthLabel.text = optionCurrentDate.stringFromFormat("MMM")
         dateLabel.text = optionStyles.showDateMonth ? optionCurrentDate.stringFromFormat("d") : nil
         yearLabel.text = optionCurrentDate.stringFromFormat("yyyy")
-        rangeStartLabel.text = optionCurrentDateRange.start.stringFromFormat("d' 'MMM' 'yyyy")
-        rangeEndLabel.text = optionCurrentDateRange.end.stringFromFormat("d' 'MMM' 'yyyy")
+
+        if (selCurrrent.showTime) {
```
Comment (ygweric) [inline Sources/WWCalendarTimeSelector.swift]: The **If** block in Swift doesn't need **()**

File: Sources/WWCalendarTimeSelector.swift [hunk 4]
```diff
@@ -1187,8 +1225,17 @@ open class WWCalendarTimeSelector: UIViewController, UITableViewDelegate, UITabl
...
+            if (optionSelectionType == .range) {
+                shouldResetRange = false;
```
Comment (ygweric) [inline Sources/WWCalendarTimeSelector.swift]: The **;** at the end of line is **NO** necessary, please search and remove them

File: Sources/WWCalendarTimeSelector.swift (hunk 1 above)
Comment (MilkBiscuit) [reply Sources/WWCalendarTimeSelector.swift]: done

File: Sources/WWCalendarTimeSelector.swift (hunk 3 above)
Comment (MilkBiscuit) [reply Sources/WWCalendarTimeSelector.swift]: done

File: Sources/WWCalendarTimeSelector.swift (hunk 4 above)
Comment (MilkBiscuit) [reply Sources/WWCalendarTimeSelector.swift]: done

File: Sources/WWCalendarTimeSelector.swift (hunk 2 above)
Comment (MilkBiscuit) [reply Sources/WWCalendarTimeSelector.swift]: done

**Pull request merged**